### PR TITLE
Hero Page

### PR DESCRIPTION
**Pull request by JosephDeng317**
Hero with animated 3D model, scroll lock, and animated text. The mobile view could be improved but works for now

### CODE REVIEW

File: src/content/docs/hero.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        <div className="grid grid-cols-1 w-full place-items-center">
+          <div className="aspect-video flex justify-center mt-30 md:mt-10 w-3/4">
+            <iframe
+              className="w-full h-full"
+              src="https://www.youtube.com/embed/tgbNymZ7vqY"
+              allowFullScreen
+            ></iframe>
+          </div>
+        </div> 
+      </div>
```
Comment (IcedTea2K) [inline src/content/docs/hero.jsx]: You can remove this part, as Serena's already delivered her PR.

**Comment by JosephDeng317** [comment]
https://github.com/user-attachments/assets/4b8ffca7-f916-493e-af5b-ed8ae610b34b

File: src/content/docs/hero.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import ThreeJSComponent from "./threejs.jsx";
+
+
+// The Hero component
+export default function Hero() {
+    return(
+      <div className="flex flex-col items-center justify-center mt-5">
+        <ThreeJSComponent/>
+
+        {/* Centered YouTube Iframe */}
```
Comment (CarsonKiibi) [inline src/content/docs/hero.jsx]: Hero should not contain youtube video as well, I believe serena implemented the video frame already

File: src/content/docs/hero.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        <div className="grid grid-cols-1 w-full place-items-center">
+          <div className="aspect-video flex justify-center mt-30 md:mt-10 w-3/4">
+            <iframe
+              className="w-full h-full"
+              src="https://www.youtube.com/embed/tgbNymZ7vqY"
+              allowFullScreen
```
Comment (CarsonKiibi) [inline src/content/docs/hero.jsx]: allowFullScreen seems to break the model size. To replicate, full screen the youtube video and get out of fullscreen. For me the fridge got smaller and squeezed (narrower on x axis)

File: src/content/docs/hero.jsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// Import necessary libraries
```
Comment (CarsonKiibi) [inline src/content/docs/hero.jsx]: Components should be in a src/components folder (see main branch or carson/footer branch).

**Review by CarsonKiibi — CHANGES_REQUESTED** [review]
Looks great and the component appears to be working as intended other than the video issue I mentioned (perhaps it only happens on my device).

The main source of unexpected behavior comes from the video for some reason, I'm not sure why but I assume the 3js component is trying to resize based on the space the fullscreen video takes up and just doesn't revert. I think serena already implemented the video component so I'd assume you can just remove it and when we add the final one we have to ensure fullscreen is disabled or maybe we can find a fix for the fullscreen issue.

Just consider adding a components folder to match the other components. I'll let Minh review too and make sure it's all good. We likely won't have time to learn 3js so we probably won't be able to provide a comprehensive review on the 3js parts but I can say it looks really nice :)

**Review by IcedTea2K — APPROVED** [review]
;-;

**Pull request merged**